Kitodo/src/main/java/org/kitodo/production/forms/createprocess/ProcessDataTab.java

```diff
@@ -188,27 +202,75 @@ void setAllDocTypes(List<SelectItem> allDocTypes) {
      * Generate process titles and other details.
      */
     public void generateProcessTitleAndTiffHeader() {
-        List<ProcessDetail> processDetails = this.createProcessForm.getProcessMetadataTab().getProcessDetailsElements();
-        Process process = this.createProcessForm.getMainProcess();
+        List<ProcessDetail> processDetails = createProcessForm.getProcessMetadataTab().getProcessDetailsElements();
+        Process process = createProcessForm.getMainProcess();
         try {
-            StructuralElementViewInterface docTypeView = createProcessForm.getRulesetManagement().getStructuralElementView(
-                docType, createProcessForm.getAcquisitionStage(), createProcessForm.getPriorityList());
+            StructuralElementViewInterface docTypeView = createProcessForm.getRulesetManagement()
+                    .getStructuralElementView(docType, createProcessForm.getAcquisitionStage(),
+                        createProcessForm.getPriorityList());
             String processTitle = docTypeView.getProcessTitle().orElse("");
             if (processTitle.isEmpty()) {
                 Helper.setErrorMessage("newProcess.titleGeneration.creationRuleNotFound",
                     new Object[] {getDocTypeLabel(docType), process.getRuleset().getTitle() });
             }
-            this.atstsl = ProcessService.generateProcessTitle(this.atstsl, processDetails,
-                processTitle, process);
+
+            String currentTitle = TitleGenerator.getCurrentValue(TitleGenerator.TITLE_DOC_MAIN, processDetails);
+
+            if ((Objects.isNull(currentTitle) || currentTitle.isEmpty())) {
+                Process parentProcess = createProcessForm.getTitleRecordLinkTab().getTitleRecordProcess();
+
+                if (Objects.nonNull(parentProcess)) {
+
+                    // get title of workpiece property "Haupttitle" from chosen title record process
+                    currentTitle = parentProcess.getWorkpieces().stream().filter(property -> "Haupttitel".equals(property.getTitle())).findFirst().map(Property::getValue).orElse(currentTitle);
```

Comment:
"Haupttitel" hier finde ich noch nicht so optimal, da der Text über die Werkstückeigenschaften angepasst werden kann und dann u.U. nicht mehr funktioniert.

Reply:
The ruleset allows to specify use attributes on the key definitions. Here, the keys indicated with `use="authorLastName"` and `use="title"` should be used as input to generate the ATS or TSL. The result should be written to any key defined as `use="processTitle"`.

https://github.com/kitodo/kitodo-production/blob/master/Kitodo/rulesets/ruleset.xsd#L416-L457

Reply:
To get the key names:

```java
RulesetManagementInterface rulesetManagement = ServiceManager.getRulesetManagementService();
String rulesetDir = ConfigCore.getParameter(ParameterCore.DIR_RULESETS);
String rulesetFullPath = Paths.get(rulesetDir, process.getRuleset().getFile()).toString();
rulesetManagement.load(new File(rulesetFullPath));
Collection<String> authorLastNameKeys = rulesetManagement.getFunctionalKeys(FunctionalMetadata.AUTHOR_LAST_NAME);
Collection<String> titleKeys = rulesetManagement.getFunctionalKeys(FunctionalMetadata.TITLE);
Collection<String> processTitleKeys = rulesetManagement.getFunctionalKeys(FunctionalMetadata.PROCESS_TITLE);
```

Reply:
> The ruleset allows to specify use attributes on the key definitions. Here, the keys indicated with `use="authorLastName"` and `use="title"` should be used as input to generate the ATS or TSL. The result should be written to any key defined as `use="processTitle"`.
> 
> https://github.com/kitodo/kitodo-production/blob/master/Kitodo/rulesets/ruleset.xsd#L416-L457

The processTitle will be written in ProcessService -> generateProcessTitleAndGetAtstsl method. These input will be still used (see TitleGenerator -> evaluateAdditionalDetailsRows).